### PR TITLE
feat: Add the TP wearables schema

### PR DESCRIPTION
**Pull request by LautaroPetaccio**
This PR adds the TP wearables schema.
To do this, it refactors the Wearable schema into a `BaseWearable` schema that is not exported and it's used to build the older `Wearable` and the new `TPWearable` types and schemas.

### CODE REVIEW

**Review by nachomazzara — COMMENTED** [review]
As far as I know, we are not going to have an entity for TPWearables, we will use the wearable entity. 

Is it ok to have a new "entity schema" for TP wearables or we should add the new fields as optional in the wearable schema? CC @LautaroPetaccio  @nicosantangelo @pentreathm @guidota

**Comment by guidota** [comment]
> As far as I know, we are not going to have an entity for TPWearables, we will use the wearable entity.
> 
> Is it ok to have a new "entity schema" for TP wearables or we should add the new fields as optional in the wearable schema? CC @guidota

we have some open questions here and added some comments with @LautaroPetaccio in https://github.com/decentraland/adr/discussions/45#discussioncomment-2199343

@nachomazzara

**Review by guidota — APPROVED** [review]

File: src/platform/wearables/base-wearable.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
+import {
+  generateValidator,
+  JSONSchema,
+  ValidateFunction
+} from '../../validation'
+// import { Rarity } from '../../dapps/rarity'
```
Comment (nachomazzara) [inline src/platform/wearables/base-wearable.ts:6]: Oops

**Review by nachomazzara — COMMENTED** [review]
Once you remove the commented line, feel free to merge it :) Great work!

**Pull request merged**